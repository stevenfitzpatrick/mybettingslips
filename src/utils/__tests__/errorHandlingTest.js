import { handleGraphQLError } from '../';

describe('Error Handling', () => {
  const generic = {
    graphQLErrors: []
  };

  const error = {
    graphQLErrors: [
      {
        functionError: {
          field: 'name',
          message: 'not found',
          title: 'this is an error'
        }
      }
    ]
  };

  test('should handle empty error', () => {
    expect(handleGraphQLError(null)).toMatchSnapshot();
  });

  test('should handle generic error', () => {
    expect(handleGraphQLError(generic)).toMatchSnapshot();
  });

  test('should handle full error', () => {
    const errorDetail = handleGraphQLError(error);
    expect(errorDetail).toMatchSnapshot({
      title: expect.any(String)
    });
    expect(errorDetail).toEqual(expect.any(Object));
    expect(errorDetail).toEqual({
      title: expect.any(String),
      message: expect.any(String),
      field: expect.any(String)
    });
  });

  const cases = [
    [
      {
        graphQLErrors: [
          {
            functionError: {
              field: 'name',
              message: 'not found',
              title: 'this is an error'
            }
          }
        ]
      },
      {
        field: 'name',
        message: 'not found',
        title: 'this is an error'
      }
    ]
  ];

  test.each(cases)('should do something', (input, expected) => {
    expect(handleGraphQLError(input)).toEqual(expected);
  });
});
