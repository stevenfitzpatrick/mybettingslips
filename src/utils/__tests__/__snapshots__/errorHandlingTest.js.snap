// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Handling should handle empty error 1`] = `undefined`;

exports[`Error Handling should handle full error 1`] = `
Object {
  "field": "name",
  "message": "not found",
  "title": Any<String>,
}
`;

exports[`Error Handling should handle generic error 1`] = `
Object {
  "message": "Generic Error has occured",
  "title": "Whhoooppss",
}
`;
